porcupine/plugins/run/no_terminal.py

```diff
@@ -185,8 +191,22 @@ def stop(self, *, quitting: bool = False) -> None:
 
         else:
             if not quitting:
-                self._queue.put(("error", "Killed."))
-                self._flush_queue()
+                assert self._thread is not None
```

Comment:
If the thread has stopped for some unexpected reason, it's better to error here than to freeze the GUI in `.get(block=True)` below.
```suggestion
                assert self._thread is not None and self._thread.is_alive()
```